extensions/objects/templates/SetOfHtmlStringEditor.js

```diff
@@ -27,7 +27,9 @@ oppia.directive('setOfHtmlStringEditor', [
         $compile(element.contents())(scope);
       },
       restrict: 'E',
-      scope: true,
+      scope: {
+        value: '='
```

Comment:
You'll also need to have `initArgs` args in here.

Reply:
In fact, any file that has the comment in line 15-17 should have `initArgs`

Reply:
@kevinlee12: I had a doubt. This directive does not uses $scope.$parent.initArgs. Instead, it uses $scope.initArgs.choices on line 46. Why does the directive then require access to the parent's initArgs?

Reply:
Also, all directives have some comment in line 15-17. Out of them, `BooleanEditor`, `CodeStringEditor`, `CoordTwoDimEditor`, `FilepathEditor`, `FractionEditor`, `GraphEditor`, `GraphPropertyEditor`, `HtmlEditor`, `ImageWithRegionsEditor`, `IntEditor`, `ListOfUnicodeStringEditor`, `LogicErrorCategoryEditor`, `LogicQuestionEditor`, `MathLatexStringEditor`, `MusicPhraseEditor`, `NonnegativeIntEditor`, `ParameterNameEditor`, `RealEditor` and `SetOfUnicodeStringEditor` do not use `initArgs` anywhere in the directive. Therefore we do not certainily require `initArgs` for them. For other directives like `NormalizedStringEditor`, `SanitizedUrlEditor`, the `scope` object already has `initArgs`.

Reply:
@kevinlee12: Regarding automatic referencing, I had a doubt too. If that was the case, why do we require such statements as `$scope.alwaysEditable = $scope.$parent.alwaysEditable` when the directive can implicitly refer to its parent's properties?
Sorry for asking so many doubts -- I had checked the directives' functioning yesterday and did not face any problem at that point. So I am a bit concerned about anything that I might have missed. Thanks :)

Reply:
For the interactions, I would recommend manually going through them in case there's something down the line that requires it (which may not be obvious from just reading the code). As for the the statements with `$scope.alwaysEditable = ...`, I think you caught the refactoring midway (since there were efforts to make things more obvious). The best way to make sure that you didn't miss anything would be to run though creating and playing an exploration manually. I'll also do the same during the reviews too. Hope that helps!

Reply:
Okay, I will test them once again and then report back. Thanks!

Reply:
So I tested this (and others) manually by running the development server and creating an exploration using them while simultaneously looking at the browser console.
In case of this directive, I did a console.log for `$scope.initArgs`. I found out that if we set `scope: true`, $scope.initArgs implicitly inherits the corresponding value of the parent but if the scope is set to:
```
scope: {
init: '@',
value: '='
}
```
$scope.initArgs remains undefined until `$scope.initArgs = $scope.$parent.initArgs` is done.
Therefore I have added this statement. Same is the case with `RandomSelector` directive. I have manually tested this approach and it seems to work fine.
Thanks!